### PR TITLE
Add asserts to DiagnosticSource

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs [hunk 1]
```diff
@@ -262,24 +262,24 @@ private Task InvokeAllResourceFiltersAsync()
 
                     await item.FilterAsync.OnResourceExecutionAsync(_resourceExecutingContext, InvokeResourceFilterAwaitedAsync);
 
-                    _diagnosticSource.AfterOnResourceExecution(_resourceExecutedContext, item.FilterAsync);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs]: I actually found another bug of the same variety by adding these asserts. If an async filter short-circuits, then `_resourceExecutedContext` hasn't been instantiated yet.

**Comment by javiercn** [comment]
:shipit:

**Comment by pranavkm** [comment]
LGTM

**Pull request closed without merging**